src/lib/Simplify.hs

```diff
@@ -729,7 +729,6 @@ simplifyOp op = case op of
     return $ Con $ Lit $ Int32Lit $ fromIntegral val
   CastOp (BaseTy (Scalar Word32Type)) (Con (Lit (Word64Lit val))) ->
     return $ Con $ Lit $ Word32Lit $ fromIntegral val
-  CastOp IdxRepTy (Con (FinVal _ i)) -> return i
```

Comment:
Is there a corresponding simplification for newtypes that should appear here?

Reply:
Nope. We no longer need to explicitly cast the payload out of a newtype, because I made it so that it can be a projection (`ProjectElt [0] newtype_atom_name` is an atom that represents the payload).